Java/src/main/java/com/gildedrose/factory/ItemEnum.java

```diff
@@ -0,0 +1,73 @@
+package com.gildedrose.factory;
+
+import com.gildedrose.Item;
+import com.gildedrose.constant.QualityConstant;
+
+import static com.gildedrose.utils.GildedRoseUtils.calculateQuality;
+
+/**
+ * 定義有特殊規則的 item enum，其他預設為 NORMAL
+ */
+public enum ItemEnum {
+    AGED_BRIE("Aged Brie") {
+        @Override
+        public void updateQuality(Item item) {
+            int amount = 1;
+            amount = item.sellIn > 0 ? amount : amount * 2;
+            item.quality = calculateQuality(item.quality, amount);
+        }
+    }, SULFURAS("Sulfuras, Hand of Ragnaros") {
+        @Override
+        public void updateQuality(Item item) {
+            // ignore
+        }
+
+        @Override
+        public void updateSellIn(Item item) {
+            // ignore
+        }
+    }, BACKSTAGE("Backstage passes to a TAFKAL80ETC concert") {
+        @Override
+        public void updateQuality(Item item) {
+            if (item.sellIn > 10) {
+                item.quality = calculateQuality(item.quality, 1);
+            } else if (5 < item.sellIn && item.sellIn <= 10) {
+                item.quality = calculateQuality(item.quality, 2);
+            } else if (0 < item.sellIn && item.sellIn <= 5) {
```

Comment:
因為是用 else if，後面的條件 always true
```suggestion
            } else if (5 < item.sellIn) {
                item.quality = calculateQuality(item.quality, 2);
            } else if (0 < item.sellIn) {
```

Reply:
感謝！ 沒注意到